services/core/shared/core/delegates.js

```diff
@@ -119,8 +132,113 @@ const getTotalNumberOfDelegates = async (params = {}) => {
+	const allNextForgersAddressList = nextForgers.map(forger => forger.account.address);
+	const activeNextForgersList = allNextForgersAddressList.slice(0, numActiveForgers);
+
+	const verifyIfPunished = delegate => {
+		const isPunished = delegate.pomHeights
+			.some(pomHeight => pomHeight.start <= lastestBlock.height
+				&& lastestBlock.height <= pomHeight.end);
+		return isPunished;
+	};
+
+	if (sdkVersion >= 4) allDelegates.sort(delegateComparator);
+	allDelegates.map((delegate, index) => {
+		if (sdkVersion < 4) {
+			if (activeNextForgersList.includes(delegate.account.address)) {
+				delegate.status = delegateStatus.ACTIVE;
+			} else delegate.status = delegateStatus.STANDBY;
+		} else {
+			delegate.rank = index + 1;
+			if (!delegate.isDelegate) delegate.status = delegateStatus.NON_ELIGIBLE;
+			else if (delegate.isBanned) delegate.status = delegateStatus.BANNED;
+			else if (verifyIfPunished(delegate)) delegate.status = delegateStatus.PUNISHED;
+			else if (activeNextForgersList.includes(delegate.account.address)) {
+				delegate.status = delegateStatus.ACTIVE;
+			} else delegate.status = delegateStatus.STANDBY;
+		}
+
+		return delegate;
+	});
+
+	if (allDelegates.length) await db.writeBatch(allDelegates);
+};
+
+const getNextForgers = async params => {
+	let forgers = {
+		data: [],
+		meta: {},
+		links: {},
+	};
+
+	try {
+		if (nextForgers.length) {
+			const offset = params.offset || 0;
+			const limit = params.limit || 10;
+
+			forgers.data = nextForgers.slice(offset, offset + limit);
+
+			forgers.meta.count = forgers.data.length;
+			forgers.meta.offset = offset;
+			forgers.meta.total = nextForgers.length;
+		} else throw new Error('Request Next Forgers data from Lisk Core');
+	} catch (err) {
+		logger.debug(err.message);
+
+		forgers = await coreApi.getNextForgers(params);
+		forgers.data = await BluebirdPromise.map(
+			forgers.data,
+			async forger => (await getDelegates({ address: forger.address })).data[0],
+			{ concurrency: forgers.data.length });
+		forgers.data.sort(delegateComparator);
+	}
+	return forgers;
+};
+
+const loadAllNextForgers = async (forgersList = []) => {
+	const limit = 100;
+	const response = await getNextForgers({ limit, offset: forgersList.length });
+	forgersList = [...forgersList, ...response.data];
+
+	if (response.data.length === limit) loadAllNextForgers(forgersList);
+	else {
+		nextForgers = forgersList;
+		computeDelegateRankAndStatus(); // Necessary to immediately update the delegate status
+		nextForgers = await BluebirdPromise.map(
+			forgersList,
+			async forger => (await getDelegates({ address: forger.account.address })).data[0],
+			{ concurrency: forgersList.length },
+		); // Update local in-mem cache with the updated delegate status information
+		logger.info(`Initialized/Updated next forgers cache with ${forgersList.length} delegates.`);
+	}
+};
+
+const reloadNextForgersCache = () => loadAllNextForgers();
+
+const reload = () => {
+	loadAllDelegates();
```

Comment:
Make use of `requestAll.js` inside `loadAllDelegates ()` to retrieve all delegates at one call.

Reply:
Done!